src/linear_algebra/general_linear_group.lean

```diff
@@ -0,0 +1,157 @@
+/-
+Copyright (c) 2021 Chris Birkbeck. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Birkbeck
+-/
+import linear_algebra.matrix
+import linear_algebra.matrix.nonsingular_inverse
+import linear_algebra.special_linear_group
+import linear_algebra.determinant
+import algebra.ordered_ring
+import ring_theory.subsemiring
+import group_theory.subgroup
+
+/-!
+# The General Linear group $GL(n, R)$
+
+This file defines the elements of the General Linear group `general_linear_group n R`,
+consisting of all invertible `n` by `n` `R`-matrices.
+
+
+## Main definitions
+
+* `matrix.general_linear_group` is the type of matrices over R which are units in the matrix ring.
+* `matrix.GL_plus.GL_pos` gives the subgroup of matrices with
+  positive determinant (over a linear ordered ring).
+
+## Tags
+
+matrix group, group, matrix inverse
+-/
+
+namespace matrix
+universes u v
+open_locale matrix
+open linear_map
+
+--TODO: Move this somewhere else.
+/--Submonoid of positive elements of an ordered semiring-/
+def pos_submonoid {R : Type*} [ordered_semiring R] [nontrivial R] : submonoid R :=
+{ carrier := {x | 0 < x},
+  one_mem' := show (0 : R) < 1, from zero_lt_one,
+  mul_mem' := λ x y (hx : 0 < x) (hy : 0 < y), mul_pos hx hy }
+
+/--The subgroup of positive units of a linear ordered commutative ring-/
+def units.pos_subgroup {R : Type*} [linear_ordered_comm_ring R] [nontrivial R] :
+subgroup (units R) :=
+{ carrier := {x | (0 : R) < x},
+  inv_mem' := λ x (hx : (0 : R) < x), (zero_lt_mul_left hx).mp $ x.mul_inv.symm ▸ zero_lt_one,
+  ..pos_submonoid.comap (units.coe_hom R)}
```

Comment:
Just as a reminder, we probably need to work out a place to put this before merging, ([Zulip thread](https://leanprover.zulipchat.com/#narrow/stream/217875-Is-there.20code.20for.20X.3F/topic/The.20submonoid.20of.20positive.20elements/near/247798909))

Reply:
It seems the suggestion was to put it in a new file. Should we do that? Can we put it in the same place as ordered_ring?

Reply:
Either a new file or `ordered_ring` sounds fine to me - can you do one of those before merging?

Reply:
It seems `algebra.ordered_ring` no longer exists. Which is why I think the build failed. I could make a new file, but do you have any ideas of where to put this file or what it should be called?

Reply:
Ah I just found `order.ring` I think thats a good place to put it.